tensorflow/core/graph/mkl_layout_pass_test.cc

```diff
@@ -185,308 +226,375 @@ TEST_F(MklLayoutPassTest, Basic) {
+  }
+REGISTER_TEST_ALL_TYPES(NodeMerge_Conv2DBackpropFilterFusion_Negative1);
+#undef REGISTER_TEST
 
 // BiasAddGrad fusion, but nodes do not match criteria for fusion.
 // Different input formats.
-TEST_F(MklLayoutPassTest, NodeMerge_Conv2DBackpropFilterFusion_Negative2) {
-  InitGraph(
-      "node { name: 'A' op: 'Input'}"
-      "node { name: 'B' op: 'Int32Input'}"
-      "node { name: 'C' op: 'Input'}"
-      "node { name: 'D' op: 'Conv2DBackpropFilter'"
-      " attr { key: 'T'                value { type: DT_FLOAT } }"
-      " attr { key: 'data_format'      value { s: 'NCHW' } }"
-      " attr { key: 'use_cudnn_on_gpu' value { b: false } }"
-      " attr { key: 'strides'          value { list: {i: 1, i:1, i:1, i:1} } }"
-      " attr { key: 'padding'          value { s: 'SAME' } }"
-      " attr { key: 'dilations'        value { list: {i: 1, i:1, i:1, i:1} } }"
-      " input: ['A', 'B', 'C'] }"
-      "node { name: 'E' op: 'BiasAddGrad'"
-      " attr { key: 'T'                value { type: DT_FLOAT } }"
-      " attr { key: 'data_format'      value { s: 'NHWC' } }"
-      " input: ['A'] }");
-  EXPECT_EQ(DoMklLayoutOptimizationPass(),
-            "A(Input);B(Int32Input);C(Input);"
-            "D(_MklConv2DBackpropFilter);DMT/_0(Const);DMT/_1(Const);"
-            "DMT/_2(Const);E(BiasAddGrad)|A->D;A->E;A:control->DMT/_0:control;"
-            "A:control->DMT/_1:control;A:control->DMT/_2:control;B->D:1;C->D:2;"
-            "DMT/_0->D:3;DMT/_1->D:4;DMT/_2->D:5");
-}
+#define REGISTER_TEST(NAME, T, INPUT)                                        \
+  TEST_F(MklLayoutPassTest, NAME##_##T) {                                    \
+    InitGraph("node { name: 'A' op: '" #INPUT "'}"                           \
+              "node { name: 'B' op: 'Int32Input'}"                           \
+              "node { name: 'C' op: '" #INPUT "'}"                           \
+              "node { name: 'D' op: 'Conv2DBackpropFilter'"                  \
+              " attr { key: 'T'                value { type: " #T " } }"     \
+              " attr { key: 'data_format'      value { s: 'NCHW' } }"        \
+              " attr { key: 'use_cudnn_on_gpu' value { b: false } }"         \
+              " attr { key: 'strides'          value { list: {i: 1, i:1, "   \
+              "i:1, i:1} } }"                                                \
+              " attr { key: 'padding'          value { s: 'SAME' } }"        \
+              " attr { key: 'dilations'        value { list: {i: 1, i:1, "   \
+              "i:1, i:1} } }"                                                \
+              " input: ['A', 'B', 'C'] }"                                    \
+              "node { name: 'E' op: 'BiasAddGrad'"                           \
+              " attr { key: 'T'                value { type: " #T            \
+              " } }"                                                         \
```

Comment:
Nit: This could be merged with the previous line.

Reply:
Done